### ISSUE TITLE
TAKING BACK appears on exclusions the reviewer never touched

### ISSUE THREAD

**Issue by itravers** [MarineAppliedResearch/MARP_API#82]
Reported from use, 2026-09-06. Phase 1 of #68.

## What happens

In **Training Data Review**, promote a page, then go back to the previous page. An
observation that is excluded on the record shows the **TAKING BACK** badge, without the
reviewer having touched it.

## What should happen

TAKING BACK means *the record still carries the exception and this reviewer has just
removed the mark, and nothing is written until the next commit*. It is a statement about
something the reviewer did. It must appear **only** when the reviewer clicks a tile that
was excluded — never on arrival at a page.

An excluded observation the reviewer has not touched should arrive **marked**, the way any
existing exception does. That is `page.seedMarks`, and the rule in
`frontend/apps/marp-mosaic-review/CLAUDE.md` is explicit: *"A page arrives with its existing
exceptions already marked. Without that, committing a page holding flags that nobody
touched silently cleared them."*

So this is not only a cosmetic mislabel. A tile that reads TAKING BACK is unmarked, and
**the next commit on that page will promote it** — silently reversing an exclusion nobody
asked to reverse. Training exclusions exist to keep bad samples out of a training set, so
the failure mode is a model trained on data somebody had already rejected.

## Where to start

The three pieces that decide this, in
`frontend/apps/marp-mosaic-review/`:

- `src/model/page.js` — `seedMarks(marks, touched, rows, isException)` skips any id in
  `touched`, and `marksAfterCommit` decides what stays marked once a commit lands.
- `src/store.js` — `refresh()` re-seeds after every query, and the commit path replaces
  `state.marks` with `marksAfterCommit(...)`.
- `src/ui/tile.js` — where TAKING BACK is derived: record carries the exception, no mark.

**Leading suspect, not yet confirmed.** `clearMarks()` and `markAllOnPage()` both do this:

```js
state.rows.forEach((r) => state.touched.add(r.observation_id));
```

Every row on the page goes into `touched`, and `touched` is *never* re-seeded, by design.
So one press of `C` (clear marks) — or one "Flag all on page" — appears to poison every
observation on that page for the rest of the session: no exception among them can ever
arrive marked again, and each will therefore read as TAKING BACK on every return visit.

That is a plausible cause and it fits the symptom, but it has **not** been reproduced
deliberately. Confirm it before fixing it, and if it is right, the question to answer is
what `touched` is actually for: it exists so that *taking a flag off and paging away does
not put it back*, which is about one id the reviewer acted on, not about every id that
happened to be on screen when they pressed a page-level key.

## Verifying the fix

The rules live in `model/`, so most of this belongs in `tests/unit/` where it costs
nothing to run. But **the badge itself is a rendering claim** and can only be proved in
`tests/e2e/render.spec.mjs` — the store has been correct every time a rendering defect
shipped here.

At minimum:

- promoting a page and returning to it leaves untouched exclusions **marked**, not taking
  back;
- pressing `C` on a page does not stop that page's exclusions being re-seeded later;
- TAKING BACK still appears when the reviewer *does* click an excluded tile;
- a commit after returning to a page does not promote an exclusion nobody touched.

Reproduce it first and watch the test fail, then fix. This is the class of defect that has
been reported twice here before, because the first fix was verified at a tier that
structurally could not see it.

**Comment by itravers** [comment]
Monday.com tracking: https://mare658599.monday.com/boards/18422190867/pulses/12993333797

**Comment by itravers** [comment]
## Reading of the cause, from #85's investigation

#85 touched the same derivation in `ui/tile.js`, so the interaction was checked deliberately.

**The two do not interact as #85 was built.** `takingBack` still requires
`state.touched.has(id)` plus *this mode's* exception on the record or the outcome, and #85
deliberately left `existingState` mode-scoped, so that expression is byte-identical.

That decision matters here. Under the obvious implementation of #85 — widening
`existingState` so it returns any workflow's value — another workflow's exception would
satisfy `existing === exception`, and **TAKING BACK would start appearing in modes that
cannot act on the tag at all**. The reason is now written into the code at both derivations
and into the app's `CLAUDE.md`, so this stays true.

**On this issue's own cause, the leading suspect looks right on reading.** `clearMarks()`
and `markAllOnPage()` both do:

```js
state.rows.forEach((r) => state.touched.add(r.observation_id));
```

and `touched` is never re-seeded, by design. So one press of `C`, or one "Flag all on page",
marks every row on that page as hand-decided for the rest of the session — after which no
exception among them can ever arrive marked again, and each reads as TAKING BACK on every
return visit.

**This has not been reproduced in a browser.** It is a reading of the code that fits the
symptom, nothing more. Reproduce it and watch a test fail before fixing it — the badge is a
rendering claim, and this app has a history of defects reported twice because the first fix
was verified at a tier that could not see them.

The question to answer while fixing it is what `touched` is actually for: it exists so that
taking a flag off and paging away does not put it back, which is about one id the reviewer
acted on — not about every id that happened to be on screen when they pressed a page-level
key.

**Comment by itravers** [comment]
Merged to develop in #88. Clear now resets the page to how it arrived — it drops the reviewer's own marks and take-backs, then lets the record's exceptions seed again — so TAKING BACK appears only where somebody clicked an individual tile. Reproduced first, and the fix proved by reintroducing the defect: 3 spurious TAKING BACK badges.
